flink-runtime/src/main/java/org/apache/flink/runtime/leaderretrieval/ZooKeeperLeaderRetrievalService.java

```diff
@@ -193,14 +185,20 @@ protected void handleStateChange(ConnectionState newState) {
 				break;
 			case SUSPENDED:
 				LOG.warn("Connection to ZooKeeper suspended. Can no longer retrieve the leader from " +
-					"ZooKeeper.");
+						"ZooKeeper.");
+				synchronized (lock) {
+					notifyLeaderLoss();
```

Comment:
@tillrohrmann what if a suspend happens while the leader info node content actually not changed?

It causes the task manager cannot find the job manager anymore. Because we actually filter same leader info inside `DefaultLeaderRetrievalService`

Reply:
Will if the leadership changed, it will generate a new id

Reply:
I think we trigger an explicit `ZooKeeperLeaderRetrievalDriver.retrieveLeaderInformationFromZooKeeper` after we reconnect. Do you think that this won't be enough?

Reply:
This work does no harm with FLINK-10052. I have no other concern.